### PR TITLE
User persister factory in storage unit

### PR DESCRIPTION
**Pull request by ssd04**

### CODE REVIEW

File: storageUnit/storageunit.go [hunk 1]
```diff
@@ -378,21 +374,14 @@ type ArgDB struct {
 }
 
 // NewDB creates a new database from database config
-func NewDB(argDB ArgDB) (types.Persister, error) {
+// TODO: refactor to integrate retries loop into persister factory; maybe implement persister
+//			factory separatelly in storage repo
```
Comment (iulianpascalau) [inline storageUnit/storageunit.go]: remove extra indent?

File: storageUnit/storageunit.go [hunk 2]
```diff
@@ -378,21 +374,14 @@ type ArgDB struct {
...
+func NewDB(persisterFactory PersisterFactoryHandler, path string) (types.Persister, error) {
```
Comment (iulianpascalau) [inline storageUnit/storageunit.go:384]: for extra protection (even if this is temporary) I would add a nil check for the provided persisterFactory. Maybe make the interface implement `NilInterfaceChecker` interface?

File: storageUnit/storageunit.go (hunk 1 above)
Comment (ssd04) [reply storageUnit/storageunit.go]: done

File: storageUnit/storageunit.go (hunk 2 above)
Comment (ssd04) [reply storageUnit/storageunit.go:384]: added nil check

**Review by iulianpascalau — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**